R/mergedatasetsbycase.R

```diff
@@ -2580,14 +2582,30 @@ variableLabelFromDataSets <- function(matched.names.row, data.sets,
     return("")
 }
 
-# Create `Source of cases` variable for the output file, which allows the consumer to determine which input file each variable came from.
-mergeSrc <- function(n.data.set.cases, data.set.names)
+# Create `Source of cases` variable for the output file,
+# which allows the consumer to determine which input file
+# each variable came from.
+mergeSrc <- function(n.data.set.cases, data.set.names, existing.mergesrc = NULL)
 {
+    if (!is.null(existing.mergesrc)) {
+        em.labels <- names(attr(existing.mergesrc, "labels"))
+        existing.mergesrc <- em.labels[existing.mergesrc]
+    }
+
     n.data.sets <- length(n.data.set.cases)
     result <- rep(seq_len(n.data.sets), n.data.set.cases)
+    file.names <- data.set.names
+    result <- file.names[result]
+
+    if (!is.null(existing.mergesrc))
+        result[!is.na(existing.mergesrc)] <- existing.mergesrc[!is.na(existing.mergesrc)]
+    data.set.names <- unique(result)
```

Comment:
There might be a problem if data set names are the same?

Reply:
For example, if the existing mergesrc had two data set values, but the two data sets had the same name, and/or data.set.names had data sets with the same names, or names that conflict with those in the existing mergesrc.